docs/features/async-main.md

```diff
@@ -0,0 +1,12 @@
+# Async Task Main
+## [dotnet/csharplang proposal](https://github.com/dotnet/csharplang/blob/master/proposals/async-main.md)
+
+## Technical Details
+
+* The compiler must recognize `Task` and `Task<int>` as valid entrypoint return types in addition to `void` and `int`.
+* The compiler must allow `async` to be placed on a main method that returns a `Task` or a `Task<T>` (but not void).
+* The compiler must generate a shim method `$EntrypointMain` that mimics the arguments of the user-defined main.
+  * `static async Task Main(...)` -> `static void $EntrypointMain(...)`
+  * `static async Task<int> Main(...)` -> `static int $EntrypointMain(...)`
+  * The parameters between the user-defined main and the generated main should match exactly.
+* The body of the generated main should be `return Main(args...).GetAwaiter().GetResult();`
```

Comment:
:question: Was there consideration of cancellation for this feature? For example, see the following:

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/841c0ac715f96af6022ab48f728fe610f5ec17de/StyleCop.Analyzers/StyleCopTester/Program.cs#L36-L65